File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int number, int limit, int power) 
{
    int sum = 0;
    for(int i=1; i<=number; i++)
    {
        int count = 0;
        for(int j=1; j * j <= i; j++)
        {
            if(i % j == 0)
            {
                count++;
                if(j != i/j)
                {
                    count++;
                }
            }
        }
        if(count <= limit) { sum += count; }
        else if(count > limit) { sum += power; }
    }
    return sum;
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.cs
using System;

using System;

public class Solution 
{
    public int solution(int number, int limit, int power) 
    {
        int[] countArr = new int[number];
        for(int i=1; i<=number; i++)
        {
            for(int j=1; j<=i; j++)
            {
                if(i%j == 0)
                {
                    countArr[i-1]++; 
                }
            }
        }
        
        int result = 0;
        for(int i=0; i<countArr.Length; i++)
        {
            if(countArr[i] > limit)
            {
                countArr[i] = power;
            }
            result += countArr[i];
        }
        return result;
    }
}